src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -67,10 +67,10 @@ static constexpr bool IsReadable(std::wstring_view text)
 
 namespace winrt::Microsoft::Terminal::Control::implementation
 {
-    TermControlAutomationPeer::TermControlAutomationPeer(TermControl* owner,
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::com_ptr<TermControl> owner,
                                                          const Core::Padding padding,
                                                          Control::InteractivityAutomationPeer impl) :
-        TermControlAutomationPeerT<TermControlAutomationPeer>(*owner), // pass owner to FrameworkElementAutomationPeer
+        TermControlAutomationPeerT<TermControlAutomationPeer>(*owner.get()), // pass owner to FrameworkElementAutomationPeer

```

Comment:
`*owner.get()` creates a copy of the underlying `TermControl` class, doesn't it? How does this work? 🤔

Reply:
If `owner` is a `com_ptr<ProjectedType>`, it probably just copies the owning pointer.

Reply:
`winrt::com_ptr<T>::get` returns `T*` according to https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/com-ptr#com_ptrget-function
`*owner.get()` should thus be of type `T`, aka `TermControl`. 🤔

Reply:
Oh, but, this is a silly thing. So.

It returns `T*`, and `*` returns `T&`. `winrt::implements` (which TermControl is eventually derived from) has an implicit conversion to _all projected types that it inherits_. This is an implicit conversion to, I would guess, `Xaml.UIElement` as a projected type. It captures a strong pointer through a `QueryInterface`!